src/lib/utils/error-logger.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const logger = {};
+
+const getLogData = (err) => {
+    let logData;
+    if (err.stack) {
+        logData = err.stack;
+    } else if (err.message) {
+        logData = err.message;
+    } else if (err.toString instanceof Function) {
+        logData = err.toString();
+    } else {
+        logData = err;
+    }
+
+    return logData;
+};
+
+module.exports.register = (server, options, next) => {
+
+    //logs error
+    logger.logError = (tags,err) => {
+        server.log(['error'].concat(tags), getLogData(err));
+    };
+
+    //logs and throws error
+    logger.logAndThrowError = (tags,err) => {
```

Comment:
I don't think our ebay friend's input applies here.  he advocated for defensive programming, i.e., assert valid input.  this, on the other hand, is asserting valid infrastructure, which I don't think should be a runtime evaluation, but rather a testing validation.  what are your thoughts on that?